android/src/main/java/com/getcapacitor/plugin/http/Http.java

```diff
@@ -453,17 +460,31 @@ private URL setParams(URL url, JSObject params) {
     }
   }
 
-  private void setRequestBody(HttpURLConnection conn, JSObject data, JSObject headers) throws IOException, JSONException {
+  private void setRequestBody(HttpURLConnection conn, PluginCall call, JSObject headers) throws IOException, JSONException {
     String contentType = conn.getRequestProperty("Content-Type");
 
     if (contentType != null) {
       if (contentType.contains("application/json")) {
         DataOutputStream os = new DataOutputStream(conn.getOutputStream());
-        os.writeBytes(data.toString());
+
+        JSObject jsObject;
+        JSArray jsArray = null;
+        jsObject = call.getObject("data", null);
+        if (jsObject != null) {
+          os.writeBytes(jsObject.toString());
+        } else {
+          jsArray = call.getArray("data", null);
```

Comment:
Can you find a way to not mix these two conditionals together. Maybe handle the jsObject case completely and the jsArray case completely without mixing these `if`/`else`'s which is a bit hard to follow.